bigframes/core/__init__.py

```diff
@@ -461,14 +470,37 @@ def shape(self) -> typing.Tuple[int, int]:
         width = len(self.columns)
         count_expr = self._to_ibis_expr(ordering_mode="unordered").count()
         sql = self._session.ibis_client.compile(count_expr)
-
+        api_methods_len = len(self._api_methods)
+
+        # Initialize api-methods as an empty dictionary
+        add_api_methods: Dict[str, str] = {}
+        # We capture the latest label if it is out of the length limit of labels count
+        if api_methods_len >= MAX_LABELS_COUNT:
```

Comment:
This logic seems weirdly placed. I'd expect this on Session / where we start our BQ jobs.

Reply:
Redo the logic here, and assign the key_name in `Session`.